Lab 4, 5, 6

zvladn7 opened the pull request:
@AlekseiMaslakov

zvladn7 committed:
added 1st impl of 4th lab

zvladn7 committed:
deleted import and unused field of class

zvladn7 committed:
added checks for unexpected situations

AlekseiMaslakov commented:
Здравствуйте
С таким хардкодом не потестить, например команда cd никогда не отработает, ибо проверка current.isDirectory() на всех машинах, кроме вашей, всегда будет возвращать false.
![image](https://user-images.githubusercontent.com/14160551/68599283-61843c00-04b1-11ea-8b23-b2ef04b1de8d.png)

zvladn7 committed:
changed root directory and added one more check for "cd .." command if current dir is "/"

zvladn7 committed:
moved condition into right place

zvladn7 commented:
@AlekseiMaslakov исправил.

zvladn7 committed:
added impl of lab 5

zvladn7 commented:
@AlekseiMaslakov еще 5 работу добавил в этот же реквест.

zvladn7 committed:
delete unneccesary throws

zvladn7 committed:
added impl of lab 6.1

AlekseiMaslakov commented:
4 - Стало чуть лучше, но cd всё так же не работает:
- нельзя перейти дальше 1 уровня
- и даже на 1 нельзя пройти, потому что стартовая директория "\" и к нему всегда добавляется ещё один слэш
5 - зачем нужен PropertiesReader?)

zvladn7 committed:
fixed work of cd command

zvladn7 committed:
deleted unnecessary imports

zvladn7 committed:
deleted PropertiesReader because Properties has already have Map

zvladn7 commented:
4. Исправил. Теперь можно перейти дальше, чем на 1 уровень и добавил проверку на тот случай, когда могла получиться пустая строка. Теперь должно работать корректно.
5. Не увидел, что Properties уже реализуют интерфейс Map. Соответственно, убрал  PropertiesReader и оставил только Main

zvladn7 commented:
@AlekseiMaslakov

zvladn7 commented:
Еще 6 лаба тут же с synchronized eсть

zvladn7 committed:
refactor lab 6

zvladn7 committed:
added full refactor of lab6: added impl of 6.2 (concurrent)
added inheritance of Account and Transaction classes

zvladn7 commented:
@AlekseiMaslakov добавил полностью 6 работу.

AlekseiMaslakov commented:
4 - ok
5 - ok
6 - на данном входном наборе данных падает:
![image](https://user-images.githubusercontent.com/14160551/69874706-ba830b00-12cc-11ea-9747-0235c9bb2b67.png)

zvladn7 committed:
added full refactor of lab6: added impl of 6.2 (concurrent)
added inheritance of Account and Transaction classes

zvladn7 committed:
corrected message of previous comment:
added transaction's generator to escape the exception with the wrong number of accounts because in previous versions it was just 4 all time so we cannot type another number account. Now this problem was solved

zvladn7 commented:
@AlekseiMaslakov не работало, так как был статический тестовый файл, в котором было всего 4 аккаунта для транзакций, соответственно, могло просто не существовать нужного аккаунта. Теперь создал генератор транзакций, который в зависимости от введенного числа аккаунтов генерирует их, следовательно, данная ошибка не должна появляться.
